overlord/assertstate/assertstate_test.go

```diff
@@ -2077,3 +2139,365 @@ func (s *assertMgrSuite) TestStore(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(store.Store(), Equals, "foo")
 }
+
+// validation-sets related tests
+
+func (s *assertMgrSuite) TestRefreshValidationSetAssertionsNop(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.setModel(sysdb.GenericClassicModel())
+
+	err := assertstate.RefreshValidationSetAssertions(s.state, 0)
+	c.Assert(err, IsNil)
+}
+
+func (s *assertMgrSuite) TestRefreshValidationSetAssertionsStoreError(c *C) {
+	s.fakeStore.(*fakeStore).snapActionErr = &store.UnexpectedHTTPStatusError{StatusCode: 400}
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.setModel(sysdb.GenericClassicModel())
+
+	tr := assertstate.ValidationSetTracking{
+		AccountID: "foo",
+		Name:      "bar",
+		Mode:      assertstate.Monitor,
+		Current:   1,
+	}
+	assertstate.UpdateValidationSet(s.state, &tr)
+
+	err := assertstate.RefreshValidationSetAssertions(s.state, 0)
+	c.Assert(err, ErrorMatches, `cannot refresh validation set assertions: unsuccessful bulk assertion refresh, fallback: cannot : got unexpected HTTP status code 400.*`)
```

Comment:
the error here reads a bit badly,   we need to do something about the fallback errors, when they are not detected I think